Source/WarpX.cpp

```diff
@@ -2204,6 +2209,11 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
         );
     }
 
+    const amrex::IntVect& ngA = ngEB;
+    const amrex::IntVect& A_nodal_flag = rho_nodal_flag;
+    const int ncomps_A = 2;
+    m_laser_envelope->AllocateLevelMFs(lev, ba, dm, ncomps_A, ngA, A_nodal_flag);
+
```

Comment:
You can try to call `InitData(lev)` here to check if the envelope field is filled as you expect:
```suggestion
m_laser_envelope->InitData(lev);

```